tourney/commands/teamname_command.py

```diff
@@ -0,0 +1,75 @@
+from .command import Command
+
+from tourney.teamnames import Teamnames
+from tourney.state import State
+from tourney.scores import Scores
+
+class TeamnameCommand(Command):
+  def __init__(self):
+    super(TeamnameCommand, self).__init__("teamname")
+
+  def execute(self, lookup=None):
+    state = State.get()
+    teamnames = Teamnames.get()
+
+    current_teams = state.teams()
+    current_teamnames = state.team_names()
+
+    my_teams = []
+    if len(current_teams) == 0:
+      scores = Scores.get()
+      matches_today = scores.today()
+      for m in matches_today:
+        if self.user_id() in m[1]:
+          my_teams.append(m[1])
+        if self.user_id() in m[3]:
+          my_teams.append(m[3])
+    else:
+      my_teams = [x for x in current_teams if self.user_id() in x]
+
+    response = ""
+
+    if len(my_teams) == 0:
+      response = "You can't claim a team name if you aren't on a team!"
+      return response
+
+    if len(my_teams) > 1:
+      response = "Cannot assign teamname when player appears in multiple teams."
+      return response
+
+    teamname = self.args()
+
+    if len(teamname) == 0:
+      response = "Please provide a new teamname, e.g. \"!teamname Example Teamname\""
```

Comment:
#nitpick: "teamname" -> "team name", "Teamname" -> "Team name"

Reply:
Also, I think you should check the team name argument before doing any of the other things because if none is given the other cycles are wasted.